src/main/java/bio/terra/service/filedata/FileService.java

```diff
@@ -76,9 +76,13 @@ public String ingestFile(
     String loadTag = loadService.computeLoadTag(fileLoad.getLoadTag());
     fileLoad.setLoadTag(loadTag);
     String description = "Ingest file " + fileLoad.getTargetPath();
+    Dataset dataset = datasetService.retrieveAvailable(UUID.fromString(datasetId));
     return jobService
         .newJob(description, FileIngestFlight.class, fileLoad, userReq)
         .addParameter(JobMapKeys.DATASET_ID.getKeyName(), datasetId)
+        .addParameter(
+            JobMapKeys.CLOUD_PLATFORM.getKeyName(),
+            dataset.getDatasetSummary().getStorageCloudPlatform())
```

Comment:
It seems odd to me to add a new parameter that can be derived from a parameter that's already provided. Is there a reason why the flight couldn't look up the platform itself? Do we usually add flight parameters that can be derived from each other? I wouldn't do this when writing a java method, for example.

Reply:
Ah, I actually am like this approach more and would like to more of this where we look up while prepping the flight and that reduces the amount of queries required of the flight.

Reply:
Why is it better to do the work here rather than inside the flight? It's the same amount of code either way. It seems to me that doing it in the flight is better, since we (the caller) should know as little as possible about what the flight needs to do, and what information it needs to do it.

I think the overall pattern is pretty poor though. This code is totally disconnected from the flight code, but it's dependent on the flight code's use of parameters. The parameters have no type check since you can put any type with any key and the type is lost once it's in the map.

One way forward would be to move this code into a static method inside the flight being constructed, and have this code pass in the minimal argument list. Once you're inside the flight class it seems like less of a violation of abstraction to extract the platform before creating the flight, although it does seem unnecessary and wasteful, as the flight itself already does the work of retrieving the dataset in its constructor.

Reply:
Yeah, Trevyn and I were talking about this last week.  It might be nice to introduce a notion of typed input class that's a bean that would allow us to pass in a bean of values rather than an arbitrary map.  That would require a (pretty significant) Stairway change though.

As for doing it here rather than in the flight, I think asking the caller to know the platform is not unreasonable but I do agree, the mechanism to pass in parameters is too lose and there's not a good way (short of Javadoc) to expose that the expected schema to the caller

Reply:
> Yeah, Trevyn and I were talking about this last week. It might be nice to introduce a notion of typed input class that's a bean that would allow us to pass in a bean of values rather than an arbitrary map. That would require a (pretty significant) Stairway change though.

Yes. Given that, moving this code next to the flight code that it depends on is an improvement we can make now.

> As for doing it here rather than in the flight, I think asking the caller to know the platform is not unreasonable

It would be reasonable if the platform wasn't already being passed in, which it is, via the `Dataset` argument. Having it stored in two places is duplication of data, which we want to avoid. And as I said above, the flight code is already calling `DatasetService.retrieveAvailable()`. If it's done here too, we're making two database calls for the same information on a single code path, which is wasteful. One way I can see to justify passing it separately is if the platform passed in is different than the one in `Dataset`, but that's not the case here.

Reply:
Yeah, I guess if we're really pushing more of that into the flight then that makes sense (I was thinking platform isn't being passed in but fair enough we could calculate in the flight.  Ok, will update (working through some serialization issues related to this anyways)